ballista/core/src/permit_stream/mod.rs

```diff
@@ -0,0 +1,81 @@
+use std::{
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+use arrow::{array::RecordBatch, datatypes::SchemaRef};
+use datafusion::{
+    error::DataFusionError,
+    execution::{RecordBatchStream, SendableRecordBatchStream},
+};
+use futures::{Stream, StreamExt};
+use tokio::time::Instant;
+
+pub struct PermitRecordBatchStream {
+    inner: SendableRecordBatchStream,
+    started_at: Instant,
+    on_close: Option<Box<dyn Fn(f64) + Send + 'static>>,
+
+    // used to release a permit when the stream is dropped
+    #[allow(dead_code)]
+    permit: tokio::sync::OwnedSemaphorePermit,
```

Comment:
You can just do `_permit` to avoid having to use the `#[allow(dead_code)]` annotation right?

Reply:
done